openedx/core/djangoapps/content_staging/models.py

```diff
@@ -0,0 +1,89 @@
+"""
+Models for content staging (and clipboard)
+"""
+from __future__ import annotations  # in lieu of typing.Self
+import logging
+# from typing import Self # Needs Python 3.11
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from opaque_keys.edx.django.models import LearningContextKeyField
+
+log = logging.getLogger(__name__)
+
+User = get_user_model()
+
+class StagedContent(models.Model):
```

Comment:
I know that I'm the one who pushed for having a more generic StagedContent concept, but I think there's value in also having a separate Clipboard model (that maybe has a OneToOneField to StagedContent). That way it's easier to enforce constraints like making sure there is only one clipboard per user, while StagedContent isn't limited in that way.

Reply:
And if/when we want to add more things that stage OLX later, like partial imports or templates, we have a pattern for adding metadata specific to those features.

Reply:
I did actually start with a separate OneToOneField clipboard model, but it turned out to have no additional fields, so it seemed kinda pointless. If you think I should put it in to help encourage that use in the future, I can?

Reply:
Oh actually you can still see it there, commented out at the bottom of `models.py` :)

Reply:
Oh actually you can still see it there, commented out at the bottom of `models.py` :)

Reply:
Hah, my mind totally skipped over that. 😄 Thank you.

It could also hold the original UsageKey it was copied from, as well as the hierarchy level (which I realize we could derive from the block_type, but I feel like it's worth calling out as a different thing). I'm mostly interested in the unique constraint though, honestly.